File: Data Structures & Algorithms/permutations/submission-2.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        ans =[]
        self.generate(nums,0,ans)
        return ans

    def generate(self, nums, indx, ans):
        if indx==len(nums):
            ans.append(list(nums))
            return 

        for i in range(indx,len(nums)):
            nums[i],nums[indx] = nums[indx],nums[i]
            self.generate(nums,indx+1,ans)            
            nums[i],nums[indx] = nums[indx],nums[i]

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set <Integer> hs = new HashSet();
        for(int i =0;i<nums.length;i++)
        {
            hs.add(nums[i]);
        }
        int ans = 0;
        int curr =0;
        for(int n: hs)
        {
            curr =1;
            while(hs.contains(n+1))
            {
                curr++;
                n=n+1;
            }
            ans = Math.max(curr,ans);
        }
        return ans;
    }
}
